snuba/consumers/consumer.py

```diff
@@ -425,6 +521,9 @@ def join(self, timeout: Optional[float] = None) -> None:
 
             step.join(timeout_remaining)
 
+        if self.__dead_letter_step:
+            self.__dead_letter_step.join(5.0)
```

Comment:
I don't know how much time we wanna leave for `DeadLetterStep.join()` but I'm assuming it should be a rare occasion to have messages in the dead letter topic

Reply:
Good that there's a timeout passed to the step